### DISCUSSION TITLE
Three laptops servers, only one finds Slingbox

### DISCUSSION THREAD

**Discussion by floormanager1** [GerryDazoo/Slinger#304] [General]
Frustrated is the word.  I have three laptops which I've been setting up and using as servers.  Two are on Windows 10, one on 11.  All three used to work but now, two appear not to be able to find the slingboxes.  The third is works fine.  I think it must be due to a firewall setting but I can't find it.  Any clues?  This is what I get from the Slinger

C:\Users\Paul\Desktop\Slinger-main>(
REM if code is not on your C drive switch to the correct drive letter
 REM g:
 REM use the full pathname to where your code is located
 cd slingbox
 slingbox_server.exe
 REM or if your using python use the full path to whatever python interpreter you're using.
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py
)
The system cannot find the path specified.
Version : 3.08e Running on Windows-10-10.0.22621-SP0 pid= 7820
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Streamer Running:  10 config.ini SLINGBOX 8080 8080 8388608
starting up on  port 8080 8388608
Checking for slingbox at ('192.168.1.16', 5301)
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
Error connecting to  ('192.168.1.16', 5301)
M1 No valid slingbox ip info found in config.ini
Finding Slingbox on local network. My IP Info =  fe80::f0b3:169b:46b6:3a2c%7
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  fe80::8ab8:d4af:4a05:1764%16
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  fe80::be58:915f:d4ed:7db2%8
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  fe80::2d87:b108:7432:cf75%18
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  fe80::dc70:1187:3666:9868%9
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  192.168.1.22
Finding Slingbox on local network. My IP Info =  ::1
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  127.0.0.1
Can't find a slingbox on network. Please make sure it's plugged in and connected. Check config.ini

**Comment by GerryDazoo** [comment]
The local firewall should not block outgoing connections. Error connecting to ('192.168.1.16', 5301). But you can always turn it off to verify, Ditto for why the server can't find any boxes on your local network. 
Run a "netstat -an" and make sure it is listening on TCP Port 8080
  TCP    0.0.0.0:8080           0.0.0.0:0              LISTENING

**Comment by floormanager1** [comment]
Thanks Gerry, it's not listening.  Just can't understand why 2 laptops are blocking and one isn't.  Don't want that one to go!

**Comment by floormanager1** [comment]
Even with the firewall turned off, it will not connect, although I can see the stream trying to connect (no image though)

C:\Users\Paul\Desktop\Slingbox M1\Slingbox>(
REM if code is not on your C drive switch to the correct drive letter
 REM g:
 REM use the full pathname to where your code is located
 cd slingbox
 slingbox_server.exe
 REM or if your using python use the full path to whatever python interpreter you're using.
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py
)
The system cannot find the path specified.
Version : 3.07 Running on Windows-10-10.0.22621-SP0 pid= 5820
Using config file config.ini
Connection Manager Running on port 8080 with 3 max streams....
Streamer Running:  3 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('192.168.1.16', 5301)
Reading Custom Remote definition from remote.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses (0.0.0.0)
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://127.0.0.1:8081
 * Running on http://192.168.1.22:8081 (Press CTRL+C to quit)
Error connecting to  ('192.168.1.16', 5301)
No valid slingbox ip info found in config.ini
Finding Slingbox on local network. My IP Info =  fe80::f0b3:169b:46b6:3a2c%7
Finding Slingbox on local network. My IP Info =  fe80::8ab8:d4af:4a05:1764%16
Finding Slingbox on local network. My IP Info =  fe80::be58:915f:d4ed:7db2%8
Finding Slingbox on local network. My IP Info =  fe80::2d87:b108:7432:cf75%18
Finding Slingbox on local network. My IP Info =  fe80::dc70:1187:3666:9868%9
Finding Slingbox on local network. My IP Info =  192.168.1.22
Finding Slingbox on local network. My IP Info =  ::1
Finding Slingbox on local network. My IP Info =  127.0.0.1
Exception in thread Thread-2 (streamer):
Traceback (most recent call last):
  File "threading.py", line 1016, in _bootstrap_inner
  File "threading.py", line 953, in run
  File "slingbox_server.py", line 539, in streamer
    for port in stati.keys:
TypeError: 'builtin_function_or_method' object is not iterable
11/15/2022, 21:46:37   Streaming connection from ('192.168.1.22', 51516)

**Reply by GerryDazoo** [reply]
OK. I recently ran through a similar issue with someone else. It turned out Defender was blocking the exe from network access. Can you make sure Defender anti-virus has an exclusion configured for the exe..

PS. Please run the latest code. That stati error has been fixed.You 5 revisions behind.

**Reply by sp222222** [reply]
> OK. I recently ran through a similar issue with someone else. It turned out Defender was blocking the exe from network access. Can you make sure Defender anti-virus has an exclusion configured for the exe..
> 
> PS. Please run the latest code. That stati error has been fixed.You 5 revisions behind.

Everything is still backward compatiable to 3.07? I kept mine as is pre N9 since all was working like I wanted it.

**Comment by floormanager1** [comment]
Sorry, I was trying different versions.  Yes, using new code.  Added rules, turned off firewall, still no joy Argh!!! One good thing today, I managed to un-brick my Pro-HD.

C:\Slingbox>(
REM if code is not on your C drive switch to the correct drive letter
 REM g:
 REM use the full pathname to where your code is located
 cd slingbox
 slingbox_server.exe
 REM or if your using python use the full path to whatever python interpreter you're using.
 REM "c:\Program Files\Python39\python.exe" slingbox_server.py
)
The system cannot find the path specified.
Version : 3.08e Running on Windows-10-10.0.22621-SP0 pid= 14872
Using config file config.ini
Connection Manager Running on port 8080 with 4 max streams using URL slingbox.
Streamer Running:  4 config.ini SLINGBOX 8080 8080 8388608
starting up on  port 8080 8388608
Checking for slingbox at ('192.168.1.16', 5301)
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
Error connecting to  ('192.168.1.16', 5301)
M1 No valid slingbox ip info found in config.ini
Finding Slingbox on local network. My IP Info =  fe80::f0b3:169b:46b6:3a2c%7
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  fe80::8ab8:d4af:4a05:1764%16
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  fe80::be58:915f:d4ed:7db2%8
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  fe80::2d87:b108:7432:cf75%18
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  fe80::dc70:1187:3666:9868%9
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  192.168.1.22
Finding Slingbox on local network. My IP Info =  ::1
Error binding socket to send broadcast [Errno 11001] getaddrinfo failed
Finding Slingbox on local network. My IP Info =  127.0.0.1
Can't find a slingbox on network. Please make sure it's plugged in and connected. Check config.ini
M1 Giving up. Sorry...

**Reply by GerryDazoo** [reply]
It's NOT the firewall, It's the virus protection you need to look at.

**Reply by floormanager1** [reply]
ahh okay...

**Comment by floormanager1** [comment]
Nope turned off virus protection and firewalls.  Added exceptions, still won't see Slingboxes.  Weirdly, the only thing I've done differently on these laptops was remove Slingbox software.  The other one that works I've not touched.  Going to strip out and start again...

**Reply by GerryDazoo** [reply]
If it's not binding to port 8080 per the netstat command nothing is going to work. Check your other servers to verify they are listening on port 8080 or whatever port you have assigned in [SERVER] section.

BTW, why do you have three servers?

**Comment by floormanager1** [comment]
Why three servers?  Good question.  I have a dedicated 'official' server, which is on 24/7, located in the garage.  This is the one that works!!  As it is located in the garage, I "play" with Slingers on another newish laptop, which was fine until last week, when unfortunately, I partially broke it when I picked it up with the power cord still in and dropped it, so now when it is off, the fan constantly runs.  As I need to replace the mother board in that, I've resurrected another laptop for mobile work.

Today, I'm going to disconnect from the internet, remove all blocks on the router (although the garage server is working perfectly) and start again.  Just don't understand why 2 will not load and a third will...

But  thank you for your help.

**Reply by GerryDazoo** [reply]
Just for grins can you copy the exe and give it a new name and then run that one instead. 
Turn back on your anti-virus (Windows Defender?) and run a full scan to make sure it stills likes the exe.
Or install python and run that.

**Comment by floormanager1** [comment]
Right, finally worked out what is going on.  The access control I've put in on the router to block the slingboxes from connecting to the internet, prevents my wifi laptops from connecting through Slinger.  Remove the blocks, they work.  Put the blocks in only the wired LAN server can see them.  Do not understand but now at least know!!!  Bloody Slingbox bricking our boxes!

**Reply by BobC1954** [reply]
How did you unbrick your ProHD? I'm not having any success - yet!

**Reply by floormanager1** [reply]
Well, I followed some instructions I found.  There's a Discord group which is working on the (I wish I knew what they were talking about but I'm afraid it's over my head).  [https://discord.com/channels/1032667324945281054/1041514081280069672](url)

I followed these instructions.  The box will stream with Slinger but as it was set up not connected to the internet, the horseshoe light stays on unless you're slinging.  Basically, download Slingplayer 1.5.  Disconnect from the internet.  Do a hard reset to the box (some say 60 secs, mine was around 20 secs but just held it in for 60).  Set up with Slingplayer 1.5, custom remote file.  Like I said, it will not complete the set-up but works.  

Proceed at your own risk.
Download a copy of the Slingplayer Desktop application v.1.5 (you’ll find it on the internet). With this version of Slingplayer, you do not need to be online to configure your Slingbox Solo.
You will NOT be able to download a remote control file for your connected hardware from Sling Media so you will need a copy of the remote .bin too. If you don’t have that, someone on this forum may have downloaded the library of remote .bin files that you can select from. Failing that, you may be able to create one or someone may have created one for your hardware already. The following website is a good resource:
http://www.hifi-remote.com/wiki/index.php/How_to_Create_Custom_Remote_Controls_and_use_them_on_your_Slingbox
SETUP INSTRUCTIONS
1.	Disconnect from the internet. Don’t turn off your router as you will need to be able to discover the Slingbox on your local network during setup.
2.	Install the Slingplayer application v1.5
3.	Place your remote .bin file (for example "V2000_RV.bin" or "V2000_PL.bin") in the following directory depending on your windows operating system:
X86 C:\Program Files\Sling Media\SlingPlayer\SBAV
X64 C:\Program Files (x86)\Sling Media\SlingPlayer\SBAV
The .bin file is uploaded to the Slingbox from this directory in the setup process.
4. Perform a hard reset of the Solo (there is a button on the back of the device that you push hold and release).
5. Open the SlingPlayer app on your PC. Your Slingbox should be discovered and be ready to setup. Select it and proceed to the next step.
6. Choose Settings / Setup Assistant
- select Configure Video Sources
- select the appropriate input (composite, S-video or component) then next>
- select "Satellite Receiver" (assuming it’s a satellite receiver) then next>
- select "Other" for manufacturer then next>
- select code "Custom" then enter the name of your remote. As per my example it will be "V2000" (case sensitive).
7. test and finish
REMEMBER
Do not attempt any setup nor use your Slingbox when it is exposed to the internet unless you have blocked all internet traffic to and from the Slingbox. It will get bricked as mine was. Others have provided useful posts regarding setting up firewall rules including the ports you should block.
As I said, you may wish to wait for someone else to confirm this method before you try it. I did a dry run on this procedure two weeks before Sling's servers were shut down and it worked for me.
Good luck.
 level 2
 
[Accomplished-Sign428](https://www.reddit.com/user/Accomplished-Sign428/)
 
· [9 hr. ago](https://www.reddit.com/r/slingbox/comments/ytnvdn/comment/iw6pomu/?utm_source=reddit&utm_medium=web2x&context=3)
Or you can try to find it in the Sling console log file.
https://www.reddit.com/r/slingbox/comments/ytrhu3/slingbox_admin_password_for_500_too_late/
 
level 1
 
[sp222222](https://www.reddit.com/user/sp222222/)
 
· [7 hr. ago](https://www.reddit.com/r/slingbox/comments/ytnvdn/comment/iw6x6hs/?utm_source=reddit&utm_medium=web2x&context=3)
I did all these steps on AV box I have. I was watching stream on it. I hit reset button and have bricked another box. I’m too leery to put my 500s back on at all. When I was configuring the AV i hit soft button so it would take static IP which I could ping immediately. But when in the wizard it had another IP so I immediately turned off internet on my router. I then saved the input and port and IP on slingbox. I watched stream for five mins using spm 1.5. I hit soft reset again and poof brick. I give up. I had four rules in place for the IP but nothing for the mac addr because my router doesn’t have that feature. router showed the red minus saying blocked but it’s bricked now. I have one more AV to test folks.
 
level 2
 
[catsby119](https://www.reddit.com/user/catsby119/)
 
· [5 hr. ago](https://www.reddit.com/r/slingbox/comments/ytnvdn/comment/iw7ek5d/?utm_source=reddit&utm_medium=web2x&context=3)
As I said in step 1. "Disconnect from the internet. Don’t turn off your router as you will need to be able to discover the Slingbox on your local network during setup".
Of course, at some stage you have to turn the internet back on and expose the Slingbox to Sling Media's "virus" so the firewall rules must be bullet proof.
In my setup, the router assigns a static IP address to all the devices on my local network, including the Slingbox, based on each device's MAC address. I can therefore set rules for the Slingbox in my firewall to:
1.	Block all ports to/from the Slingbox's IP address; and
2.	Open port "x" (where port "x" is Gerry's Slinger server on my local network) on the Slingbox IP address but only to the IP address of the PC where the Slinger server is installed.
My Solo is bricked so I cannot test these rules right now BUT as proof of concept, I have already set up a rule that I call "Local Traffic Only" that blocks all ports to and from all IP addresses. I activate that rule when I want to stop all internet traffic whilst still allowing communication between devices in my LAN (these devices already have firewall permissions for the ports they access on the internet). It works a treat. The proposed "Slingbox" rules above are a simple variation of that where I replace "all IP addresses" with just the static IP address of the Slingbox.
Assuming your router can assign static IP addresses to devices in your LAN based on MAC address, you could test my theory by using the above rules to block internet traffic to a device (say another PC or notebook or even your mobile phone) in your LAN (other than your Slingbox of course) then try communicating with it inside your LAN and outside your LAN. Then see if the device has internet access and if it can communicate with other devices in your LAN. If testing with your mobile phone disable the mobile network access first so that it has WiFi access to your LAN only.
 

 
1
 

 

 

 
 
 
level 3
 
[sp222222](https://www.reddit.com/user/sp222222/)
 
· [5 hr. ago](https://www.reddit.com/r/slingbox/comments/ytnvdn/comment/iw7ffp0/?utm_source=reddit&utm_medium=web2x&context=3)
I did get my proHD back by doing a 63 second hard press reset button. admin is password. it works with slinGER again source 3 but I cannot save settings so it uses 5001 port default
